pkg/client/client.go

```diff
@@ -297,3 +297,16 @@ func DefaultOpenShiftUserAgent() string {
 	version = seg[0]
 	return fmt.Sprintf("%s/%s (%s/%s) openshift/%s", path.Base(os.Args[0]), version, runtime.GOOS, runtime.GOARCH, commit)
 }
+
+type HybridClient struct {
```

Comment:
I am not, we should be providing an implementation of ScaleNamespacer and
ScaleInterface that multiplexes to clients based on known kinds.

On Thu, Oct 22, 2015 at 1:35 PM, David Eads <notifications@github.com>
wrote:

> In pkg/client/client.go
> <https://github.com/openshift/origin/pull/5310#discussion_r42779351>:
>
> > @@ -297,3 +297,16 @@ func DefaultOpenShiftUserAgent() string {
> >  	version = seg[0]
> >  	return fmt.Sprintf("%s/%s (%s/%s) openshift/%s", path.Base(os.Args[0]), version, runtime.GOOS, runtime.GOARCH, commit)
> >  }
> > +
> > +type HybridClient struct {
>
> See how the Kubernetes client is currently implemented:
> https://github.com/kubernetes/kubernetes/blob/071d21257fdfd439a9286fffbd5972c85643cd49/pkg/client/unversioned/client.go#L128-l134
>
> We should do the same for our client
>
> This is what they do
>
> // Client is the implementation of a Kubernetes client.type Client struct {
>     *RESTClient
>     *ExtensionsClient
>     // TODO: remove this when we re-structure pkg/client.
>     *DiscoveryClient
> }
>
> I don't think that this is a good thing. Having a single interface like
> that encourages non-specificity and it does not scale well as you add
> different clients. I think that a reasonable division is on the APIGroup
> boundary. Using different clients for different APIGroups will help keep
> the dependency structure clear, prevents a mega-client problem, and allows
> a clear expectation as APIGroups are enabled and disabled.
>
> @smarterclayton <https://github.com/smarterclayton> as I recall, you were
> not a fan combined Clients.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5310/files#r42779351>.
>